### DISCUSSION TITLE
Add dependency for airflow for python3.8 but not upper version

### DISCUSSION THREAD

**Discussion by BenediktRamsauer** [apache/airflow#40586] [Q&A]
Hi,

For one of my PR, I will use the built-in library `ast`. From `python 3.9`, the function `ast.unparse` exists but for `python 3.8`, I need to install the library `astunparse`. 

Is there an existing way to install dependencies only for `py38`? 

This way, when a version is not supported any more, there is no need to keep installing these libraries. I suppose it will be in somewhere in https://github.com/apache/airflow/blob/3583329c1b53530c4c2e551e37affe78761ed3e5/hatch_build.py#L505

Thank you in advance!

**Comment by potiuk** [comment]
https://github.com/apache/airflow/blob/3583329c1b53530c4c2e551e37affe78761ed3e5/hatch_build.py#L408 has the list of dependencies and there is (python standard) way of specifying dependencies including dependency on python version (see the list you will find example there). But there are two things:

1) You likely do not need it - we definitely do not want to add extra dependencies to Airflow and adding astunparse to do what you planned to do for https://github.com/apache/airflow/issues/40585 is not needed, because as I explained there, you can achieve the same result by adding `from future import __annotations__` without writing extra (and risky) code - especially that we do not really want to run AST parsing for that purpose only.

2) even we would like to add it - the right place to add it would be in cncf.kubernetes provider dependencies (and possibly docker provider dependencies) - and those dependencies are in `airflow/providers/PROVIDER_PATH/provider.yaml` file and dynamically retrieved from ther during hatch build and provider package builds.
